### DISCUSSION TITLE
Add options for us to support PaperWM financially

### DISCUSSION THREAD

**Discussion by distefam** [paperwm/PaperWM#723] [General]
After using various tiling window managers and switching back to Gnome, PaperWM is precisely what I've been looking for to merge the two. It is well thought out, designed, and is a comprehensive solution to tiling on Gnome. I'd love to be able to support it but don't have the time to contribute code right now so I'd like to help contribute financially and I'm sure others would as well.

A lot of other open source projects have some sort of financial donation program set up, like:
- [Liberapay](https://liberapay.com/)
- [KoFi](https://ko-fi.com/)
- [Buy me a coffee](https://www.buymeacoffee.com/)

Perhaps you might want to add one of these?

**Comment by jtaala** [comment]
Hey @distefam,

We appreciate the gesture - I'm not sure we're in a position to accept donations.  There's just a few of us developing and working on PaperWM when we can.

I'm not sure how this would work in terms of transparency and how the funds would be used.  Plus, whenever money is involved things get much more complicated it seems (just my thoughts).

**Reply by dylansumser** [reply]
Just wanted to second the sentiment here. Absolutely love Paper WM, it’s a big part of what’s sold me on Linux as my main desktop. I would also want to do what I can to help support it. I admire and respect the decision not to accept financial support right now, but still want to thank you for the great work ❤️!

**Comment by jtaala** [comment]
Hey all, so I've been thinking about this one.  

I'm not opposed to adding options to supporting the continued development and support for PaperWM financially for those that might want to do so.  I would just want to make sure it's done transparently.

More importantly, it should be done in a way that encourages more community developer contributions, e.g. [PRs](https://github.com/paperwm/PaperWM/pulls), working to address current [issues](https://github.com/paperwm/PaperWM/issues), improving documentation etc.

I personally am struggling to prioritise and work on bug fixes, new requested features, Gnome 46 compatibility, etc., amidst my day job, family stuff etc. - that's not unique though, others are in the same boat.  But any help would be appreciated by those that are able to - and allowing  others to receive a little support from donations etc. might help here.

Am seeking thoughts, input etc.  Looks like GitHub allows various ways for users to donate, but the easiest seems to be adding (up to 4) custom URL links (which lead to donations to users).  So, we could:

- provide user links on github for the latest ~~four~~ three (since EGO supports three contributors for a donation source) contributors (that contributed to a release, i.e. their PR / contribution was accepted);
- links would be added/removed on a rolling basis (i.e. latest 3 contributors to any release?);
- When accepting a PR (into `develop` branch... which will make it's way into `release`), I would ask the contributor if they have a donation link that they want added).  Link would be added to github sponsorship (`FUNDING.yml`) on merge into `release` branch;
- link ordering could be based on frequency of contribution to releases (e.g. if users contributed to multiple releases, their link would be higher in the list?);

Reasoning for the above: it seeks to reward any contribution to a release via PR, which could be a fix, feature, or improving documentation etc.  It also allows the community to pick/choose whom to donate to, e.g. maybe someone worked on and provided an obscure/niche feature they've been waiting for?

Thoughts?

**Reply by Lythenas** [reply]
If you or someone else wants to spend time setting this up, I'm fine with it. Although I feel like this is probably more work than it's worth.

I have also seen some kind of "bug bounty" program (external service) in other open source repos, where you could pay for a specifically issue and once it's fixed the money would be payed out. I think to the person that fixes the issue, but not sure. Unfortunately I can't remember where I saw it or what it was called. I also haven't looked into it more, so not sure if it is usable by us.

**Reply by jtaala** [reply]
Thanks for the thoughts @Lythenas.  We can see how it goes, but it'll be slow going I think to start.

**Reply by jtaala** [reply]
@Thesola10 and @xalt7x, given you were among the last few contibutors, and if you would like to have a donation link added, please send me the link on https://matrix.to/#/#paperwm.github.com:matrix.org (matrix.org) or if you prefer, post a comment at the bottom of https://jaytaala.com (your comment won't be viewable publicly since I need to approve that - so I'll just use it to obtain the link and not publish any comment there).

**Reply by Thesola10** [reply]
Bounty programmes like [Gitpay](https://gitpay.me) seem to be heavily oriented towards issue authors, which means a couple things:

- The bounty exists outside of the project's control. Gitpay acts like a broker between people eager to see an issue fixed, and developers fixing that issue. Could be a good fit for PaperWM's open, (theoretically) community-owned development process
- However, it only applies to issues, meaning someone who made a contribution to fix an issue they're the first to actually raise (like me!) won't be rewarded -- you can't reward existing or ongoing work with it.
- Therefore, for self-started contributions, it kinda requires "selling" people on your issue, which is really not something we want to encourage

For self-started contributions, splitting per release could solve that issue, but then there needs to be scaling among contributors to that release so I dunno.

**Reply by Thesola10** [reply]
Apart from all this, right now I think the person most deserving of PaperWM donations is you, Jay! You've been spending a significant amount of time dealing with most of the maintenance work covering PaperWM, acting as the de facto project lead, and all those hours spent working on PaperWM absolutely deserve payment. Democracy is important, but don't sell yourself short :heart:

**Comment by jtaala** [comment]
> Apart from all this, right now I think the person most deserving of PaperWM donations is you, Jay! You've been spending a significant amount of time dealing with most of the maintenance work covering PaperWM, acting as the de facto project lead, and all those hours spent working on PaperWM absolutely deserve payment. Democracy is important, but don't sell yourself short ❤️

That's much appreciated @Thesola10 - I don't think any of us do this for reward, and I still think PaperWM is something special and I hope it grows, evolves, and continues to improve over time (which I why I keep working on it! - plus, I just can't go back to normal windowing, or even, tiling now :-)).

Also, I expect (and hope) any donations would be small (there's many more worthy and good causes to donate to in the world!), but I thought they might brighten someone's day as gesture for us being grateful for any / all contributions.

**Comment by smichel17** [comment]
I've thought about this topic a lot, as part of my involvement in [Snowdrift.coop](https://snowdrift.coop), a still-under-construction (for the last decade…) crowdfunding platform specifically for FLOSS (and other public goods). But I'll keep this short so I'm not here writing all day. 

I [would not recommend a bounty platform](https://wiki.snowdrift.coop/about/existing-mechanisms#bounties). In this case, I would recommend [OpenCollective](https://opencollective.com/).

- Their 'fiscal host' system, where another organization holds the money for the project, takes a lot of the administrative burden off of a small volunteer project like this one. Specifically, I'd choose [Open Source Collective (OSC)](https://opencollective.com/opensource) as the fiscal host.
- The platform provides good tooling for transparency around how funds are spent.

---

My advice for how to spend funds is not more insightful than the average person, but here's some ideas:

- Buy a domain name, even if it just redirects to this repository.
- Buy hardware for core maintainers so they can test edge cases. E.g. if I recall, at one point @jtaala didn't have a [good] multiple-monitor setup. Not sure if that's still the case.
- If you end up with more funds and there are contributors who are contractors at their dayjob, consider a contract. Start small, for building trust.

For governance, maybe something like: spending funds requires a majority vote of the [collaborators](https://github.com/paperwm/PaperWM/settings/access) on this repo. Maybe with the project lead (currently @jtaala) having veto power?

**Comment by jtaala** [comment]
> I've thought about this topic a lot, as part of my involvement in [Snowdrift.coop](https://snowdrift.coop), a still-under-construction (for the last decade…) crowdfunding platform specifically for FLOSS (and other public goods). But I'll keep this short so I'm not here writing all day.

Thanks @smichel17 - yeah, I agree.  This is not my forte for sure, so appreciate the comments and would be happy to go along / support your thoughts / approach here (if you or anyone wanted to implement this and/or contribute to this are of the project).

I thought we'd try the current approach as it seemed easy/light, without losing our _community/volunteer vibe_ :-).  But in practice it's a bit tedious to get donation links for contributors (and there's a limit to how many we can add).

P.S. I like the domain name idea too.  Using funds for hardware directly related to the project (such as the case you mentioned) makes a lot of sense (as it contributes to the project).  Another good example of this is `touch-screen` support, since I don't have anything touch and therefore can't work on / test anything there (so I've been relying on @Thesola10 for testing and PRs related to touchscreen support there).
